contracts/ChipRegistry.sol

```diff
@@ -137,225 +124,224 @@ contract ChipRegistry is IChipRegistry, ClaimedPBT, Ownable {
-     *                              on implementation of TransferPolicy
+     * Get ERS node from tokenData and then sets the new Owner of the chip on the ERSRegistry.
      */
-    function transferToken(
-        address chipId,
-        bytes calldata signatureFromChip,
-        uint256 blockNumberUsedInSig,
-        bool useSafeTransferFrom,
-        bytes calldata payload
-    ) 
-        public
-        override(ClaimedPBT, IPBT)
-    {
-        // Validations happen in ClaimedPBT / TransferPolicy
-        ClaimedPBT.transferToken(chipId,  signatureFromChip, blockNumberUsedInSig, useSafeTransferFrom, payload);
-        _setERSOwnerForChip(chipId, msg.sender);
-    }
 
     /**
-     * @dev ONLY CHIP OWNER (enforced in ClaimedPBT): Sets the owner for a chip. Chip owner must submit transaction
-     * along with a signature from the chipId commiting to a block the signature was generated. This is to prevent
-     * any replay attacks. If the transaction isn't submitted within the MAX_BLOCK_WINDOW from the commited block
-     * this function will revert. Additionally, the chip's ERS node owner is updated to maintain state consistency.
+     * @notice Set the owner of a chip through its projectRegistrar
      *
-     * @param _chipId           The chipId to set the owner for
-     * @param _newOwner         The address of the new chip owner
-     * @param _commitBlock      The block the signature is tied to (used to put a time limit on the signature)
-     * @param _signature        The signature generated by the chipId (should just be a signature of the commitBlock)
+     * @param _chipId           The chip public key
+     * @param _newOwner         The new owner of the chip
      */
-    function setOwner(
-        address _chipId,
-        address _newOwner,
-        uint256 _commitBlock,
-        bytes calldata _signature
-    )
-        public
-        override
-    {   
-        // Validations happen in ClaimedPBT, ERC721 doesn't allow transfers to the zero address
-        ClaimedPBT.setOwner(_chipId, _newOwner, _commitBlock, _signature);
-        _setERSOwnerForChip(_chipId, _newOwner);
+    function setChipNodeOwner(
+        address _chipId, 
+        address _newOwner
+    ) 
+        external 
+    {
+        IProjectRegistrar projectRegistrar = IProjectRegistrar(msg.sender);
+
+        require(projectEnrollments[projectRegistrar].chipsAdded, "Only enrolled projects with chips can call.");
+        require(chipEnrollments[_chipId].chipEnrolled, "Chip not added");
+        require(chipEnrollments[_chipId].projectRegistrar == address(projectRegistrar), "ProjectRegistrar does not own chip");
+
+        bytes32 _nameHash = chipEnrollments[_chipId].nameHash;
+        bytes32 rootNode = projectRegistrar.rootNode();
+       
+        bytes32 chipErsNode = keccak256(abi.encodePacked(rootNode, keccak256(abi.encodePacked(_nameHash))));
```

Comment:
Why are we hashing the `_nameHash` again? Seems like that should already be hashed.

Doesn't seem like there's tests for this function def think we should have those, my guess is this would fail if we did

Reply:
Changed in WIP commit, still need test coverage to verify.